src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -143,6 +135,13 @@ export const DropdownMenu: React.FC<Props> = ({
     }
   };
 
+  const handleOnClick: MouseEventHandler = (e) => {
+    focusIndex = refs.current.list.findIndex((ref) =>
+      ref.contains(e.target as HTMLElement),
+    );
+    focusItem(focusIndex);
```

Comment:
Required since [Safari doesn't focus `<button>` elements on click](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#clicking_and_focus)

Reply:
Could you move this and the comment above into the code?